lib/deflation.cpp

```diff
@@ -170,37 +176,37 @@ namespace quda {
     std::vector<ColorSpinorField*> in_;
     in_.push_back(static_cast<ColorSpinorField*>(b_sloppy));
 
-    blas::cDotProduct(vec, rv_, in_);//<i, b>
+    blas::cDotProduct(vec.get(), rv_, in_);//<i, b>
 
     if(!param.use_inv_ritz) 
     {
+      if( param.eig_global.extlib_type == QUDA_MAGMA_EXTLIB ) {
 #ifdef MAGMA_LIB
-      magma_Xgesv(vec, param.ld, param.cur_dim, param.matProj, param.ld, sizeof(Complex));
+        magma_Xgesv(vec.get(), param.ld, param.cur_dim, param.matProj, param.ld, sizeof(Complex));
 #else
-      Map<MatrixXcd, Unaligned, DynamicStride> projm_(param.matProj, param.cur_dim, param.cur_dim, DynamicStride(param.ld, 1));
-      Map<VectorXcd, Unaligned> vec_ (vec, param.cur_dim);
-
-      VectorXcd  vec2_(param.cur_dim);
+        errorQuda("MAGMA library was not built.\n");
+#endif
+      } else {
```

Comment:
This else statement here just catches the `extlib_type != QUDA_MAGMA_EXTLIB` case and not specifically the Eigen case.  All branch-on-extlib_type statements should be made safe to throw an error if a given `extlib_type` is not supported.